contracts/StableJoeStaking.sol

```diff
@@ -40,28 +40,47 @@ contract StableJoeStaking is Initializable, OwnableUpgradeable {
          */
     }
 
-    IERC20Upgradeable public joe;
+    // @dev gap to keep the storage ordering, replace `IERC20Upgradeable public joe;`
+    uint256[1] private __gap0;
+
+    /// @notice The address of the JOE token
+    IERC20Upgradeable public immutable joe;
 
     /// @dev Internal balance of JOE, this gets updated on user deposits / withdrawals
     /// this allows to reward users with JOE
     uint256 public internalJoeBalance;
+
     /// @notice Array of tokens that users can claim
     IERC20Upgradeable[] public rewardTokens;
+
+    /// @notice Mapping to check if a token is a reward token
     mapping(IERC20Upgradeable => bool) public isRewardToken;
+
     /// @notice Last reward balance of `token`
     mapping(IERC20Upgradeable => uint256) public lastRewardBalance;
 
+    // @notice The address where deposit fees will be sent
     address public feeCollector;
-
     /// @notice The deposit fee, scaled to `DEPOSIT_FEE_PERCENT_PRECISION`
-    uint256 public depositFeePercent;
+    uint88 public depositFeePercent;
+    /// @notice Reentrancy guard
+    bool public reentrant;
```

Comment:
This only work because `depositFeePercent` is zero atm. That's a bit dangerous imo, imagine we put this upgrade on hold for some reason, then later we update the deposit fee and decide to finally carry on the upgrade while forgetting about this